domain/src/main/java/org/oppia/domain/question/QuestionDataController.kt

```diff
@@ -0,0 +1,83 @@
+package org.oppia.domain.question
+
+import android.content.Context
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import org.oppia.app.model.Question
+import org.oppia.domain.topic.TEST_TOPIC_ID_0
+import org.oppia.domain.topic.TEST_TOPIC_ID_1
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.data.DataProviders
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val QUESTION_DATA_PROVIDER_ID = "QuestionDataProvider"
+
+/** Controller for retrieving an exploration. */
+@Singleton
+class QuestionDataController @Inject constructor(
```

Comment:
Nit: per #120 this should be ``QuestionTrainingController``

Reply:
Done